ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/TextArea.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ui;
+
+import com.google.gwt.dom.client.NativeEvent;
+
+import org.eclipse.che.ide.ui.smartTree.KeyboardNavigationHandler;
+
+/**
+ * Text Area widget that supports shifting to new line by pressing Enter key.
+ *
+ * @author Igor Vinokur
+ */
+public class TextArea extends com.google.gwt.user.client.ui.TextArea {
```

Comment:
Don't use same name in class hierarchy. Maybe it would be better to rename this class smth to related with multiline. e.g. `MultilineTextArea` or `LFTextArea` (from https://en.wikipedia.org/wiki/Newline)